src/aspire/basis/fb_3d.py

```diff
@@ -176,43 +170,36 @@ def _evaluate(self, v):
                 ang = self._precomp["ang"][:, ind_ang]
                 ang_radial = np.expand_dims(ang[ang_idx], axis=1) * radial[r_idx]
                 idx = ind + np.arange(0, len(idx_radial))
-                x[mask] += ang_radial @ v[idx]
+                x[:, mask] += v[:, idx] @ ang_radial.T
                 ind += len(idx)
                 ind_ang += 1
 
             ind_radial += len(idx_radial)
 
-        x = m_reshape(x, self.sz + x.shape[1:])
-        x = roll_dim(x, sz_roll)
+        return x.reshape(*stack_shape, *self.sz)
 
-        return x.T
-
-    def _evaluate_t(self, v):
+    def _evaluate_t(self, x):
         """
         Evaluate coefficient in FB basis from those in standard 3D coordinate basis
 
-        :param v: The coefficient array to be evaluated. The first dimensions
+        :param x: The coefficient array to be evaluated. The first dimensions
             must equal `self.sz`.
         :return: The evaluation of the coefficient array `v` in the dual
             basis of `basis`. This is an array of vectors whose first dimension
             equals `self.count` and whose remaining dimensions correspond
             to higher dimensions of `v`.
         """
-        v = v.T
-        x, sz_roll = unroll_dim(v, self.ndim + 1)
-        x = m_reshape(
-            x, new_shape=tuple([np.prod(self.sz)] + list(x.shape[self.ndim :]))
-        )
-
+        stack_shape = x.shape[: -self.ndim]
```

Comment:
We keep repeating this unroll/roll pattern, which makes me wonder why we needed to remove it in the first place…?

Reply:
If you're referring to the `unroll_dim/roll_dim` methods, I removed them because they are using `m_reshape` under the hood.